src/main/java/org/eclipse/basyx/submodel/metamodel/facade/SubmodelElementMapCollectionConverter.java

```diff
@@ -151,9 +152,18 @@ public static SubmodelElementCollection mapToSmECollection(Map<String, Object> s
 			for (Object o: smElementsSet) {
 				Map<String, Object> smElement = (Map<String, Object>) o;
 				String id = (String) smElement.get(Referable.IDSHORT);
+				
+				if(smElementsMap.containsKey(id)) {
+					throw new IdShortDuplicationException(SubmodelElementMapCollectionConverter.class, (Map<String, Object>) smElementsMap);
```

Comment:
The class passed to the constructor is the class that was intended to construct. Thus, passing SubmodelElementMapCollectionConverter.class does note make any sense. At the same time, we don't know which kind of element should be created here. In consequence, please add a constructor in MetamodelCreationException that takes an error message and add one constructor in IdShortDuplicationException that takes a map and calls this super constructor with a message like "Could not construct meta model element due to IdShort duplication. Passed argument was " + map.toString());

Reply:
- Added a constructor in MetamodelCreationException that takes an error message
- Added a constructor in IdShortDuplicationException that takes a map